python/lsst/ts/observatory/control/maintel/mtcs.py

```diff
@@ -579,6 +579,101 @@ async def dome_el_in_position(self) -> str:
+
+        Raises
+        ------
+        RuntimeError
+            If a bad state is encountered or the expected state is not reached
+             in time.
+        """
+        az_motion = await self.rem.mtdome.evt_azMotion.aget(timeout=self.long_timeout)
+        while az_motion.state not in expected_states:
+            az_motion = await self.rem.mtdome.evt_azMotion.next(
+                Flush=False, timeout=timeout
+            )
+
+            if az_motion.state in bad_states:
+                raise RuntimeError(
+                    f"Dome transitioned to an invalid state:" f" {az_motion.state}"
+                )
+
+            # If check_in_position is True, verify the dome is in the correct
+            # position.
+            if check_in_position and not az_motion.inPosition:
+                self.log.debug(
+                    f"Dome not yet in position. Current state: " f"{az_motion.state}."
+                )
+
+            self.log.debug(
+                f"Dome state: {az_motion.state}, inPosition: " f"{az_motion.inPosition}"
+            )
+
+            # If we reach this point, the dome has successfully reached an
+            # expected state.
+        if check_in_position and not az_motion.inPosition:
+            raise RuntimeError(
+                f"Dome reached the expected state "
+                f"{az_motion.state} but is not yet in position."
+            )
+
+    async def park_dome(self) -> None:
+        """Park the dome by moving it to the park azimuth."""
+        self.log.info(f"Parking dome at azimuth {self.dome_park_az}.")
+
+        # Ensure the dome is ready for a move
+        await self.assert_all_enabled(
+            message="All components need to be enabled for parking the Dome."
+        )
+
+        # check first if Dome is already in PARKED state
+        az_motion = await self.rem.mtdome.evt_azMotion.aget(timeout=self.fast_timeout)
+
+        if az_motion.state == MTDome.MotionState.PARKED:
+            self.log.info("Dome is already in PARKED state.")
+        else:
+            # Move the dome to the park position
+            await self.rem.mtdome.cmd_park.start(timeout=self.long_timeout)
+
+            # Define expected and bad states for parking
+            expected_states = {MTDome.MotionState.PARKED}
+            bad_states = {
+                MTDome.MotionState.ERROR,
+                MTDome.MotionState.UNDETERMINED,
+                MTDome.MotionState.DISABLED,
+                MTDome.MotionState.DISABLING,
+            }
+
+            # Wait for the dome to emit the evt_azMotion event in the PARKED
+            # state
+            self.log.info("Waiting for dome to reach the PARKED state.")
+
+            # Wait for the dome to park, ensuring it doesn't reach an invalid
+            # state
+            await self.wait_for_dome_state(
+                expected_states,
+                bad_states,
+                timeout=self.long_timeout,
```

Comment:
I think the `long_timeout` might not be long enough for this operations. I think this might require its own timeout, something like `dome_park_timeout`? And I think this needs to be something like 5-10 minutes. I am not sure if this is enough to park the dome but I think it is a good start. Maybe add a comment when you create this attribute that this might need updating in the future.